src/ui/ResponsiveComponents/ResponsiveRecommendation.ts

```diff
@@ -173,51 +168,39 @@ export class ResponsiveRecommendation implements IResponsiveComponent {
   }
 
   private getFacetSliders(): any[] {
-    let facetSliders = [];
-    _.each(this.coveoRoot.findAll('.' + Component.computeCssClassNameForType(`FacetSlider`)), facetSliderElement => {
-      let facetSlider = Component.get(facetSliderElement);
-      facetSliders.push(facetSlider);
+    const facetSliders = [];
+    this.coveoRoot.findAll(Component.computeSelectorForType('FacetSlider')).forEach(facetSliderElement => {
+      const facetSlider = Component.get(facetSliderElement);
+      facetSlider && facetSliders.push(facetSlider);
     });
     return facetSliders;
   }
 
   private getFacets(): any[] {
-    let facets = [];
-    _.each(this.coveoRoot.findAll('.' + Component.computeCssClassNameForType('Facet')), facetElement => {
-      let facet = Component.get(facetElement);
-      facets.push(facet);
+    const facets = [];
+    this.coveoRoot.findAll(Component.computeSelectorForType('Facet')).forEach(facetElement => {
+      const facet = Component.get(facetElement);
+      facet && facets.push(facet);
     });
     return facets;
   }
 
-  private dismissFacetSearches(): void {
-    _.each(this.facets, facet => {
-      if (facet.facetSearch && facet.facetSearch.currentlyDisplayedResults) {
-        facet.facetSearch.completelyDismissSearch();
-      }
-    });
-  }
-
   private enableFacetPreservePosition(): void {
-    _.each(this.facets, facet => (facet.options.preservePosition = true));
+    this.getFacets().forEach(facet => (facet.options.preservePosition = true));
   }
 
   private disableFacetPreservePosition(): void {
-    _.each(this.facets, facet => (facet.options.preservePosition = false));
+    this.getFacets().forEach(facet => (facet.options.preservePosition = false));
   }
 
   private drawFacetSliderGraphs(): void {
-    _.each(this.facetSliders, facetSlider => facetSlider.drawDelayedGraphData());
+    this.getFacetSliders().forEach(facetSlider => facetSlider.drawDelayedGraphData());
   }
 
   private registerOnOpenHandler(): void {
     this.dropdown.registerOnOpenHandler(this.drawFacetSliderGraphs, this);
   }
 
```

Comment:
Would it make sense to add any unit tests for these changes?

Reply:
the changes wont be seen with UTs since it's all private. It's more in the order of the integration test that is has to be checked. I honestly don't know if all those facet methods are really pertinent....